[Sifan] iP

yespiggy opened the pull request:
### LilyBot
> "Let our advance worrying become advance thinking and planning." - Winston Churchill [source](https://www.inloox.com/company/blog/articles/the-62-most-inspiring-project-management-quotes/)

LilyBot frees your mind of having to remember things you intend to do. It's,
* easy to learn
* Interactive
* ~~FAST~~ ***SUPER FAST*** to use

All you need to do is,
1. download it from [here](https://github.com/yespiggy/ip/releases/tag/v0.1)
2. run `java -jar duke.jar` in your terminal
3. add your tasks
4. let it manage your tasks for you 🧚 🧚 💗  

And it is **FREE**!!

Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

--- 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) throws IOException {
        new Duke(Storage
                .getFilePathToDukeTxt())
                .run();

    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yespiggy committed:
Rename, Greet, Exit

yespiggy committed:
Level 1: Echo

yespiggy committed:
Level-2 Add, List

yespiggy committed:
Level-3 Mark as Done

yespiggy committed:
Level-4 ToDos, Events, Deadlines

yespiggy committed:
A-TextUiTesting: Automated Text UI Testing

yespiggy committed:
Level-5 Handle Errors

yespiggy committed:
Level-6 Delete

yespiggy committed:
Level-7 Save to hard disk

yespiggy committed:
Merge branch 'branch-Level-7'

yespiggy committed:
Level-8 Dates and Times

yespiggy committed:
Merge Level-8 to master

yespiggy committed:
A-MoreOOP

yespiggy committed:
moreOOP & Fix event time format

yespiggy committed:
A-Packages, duke

yespiggy committed:
A-Packages, duke

yespiggy committed:
Merge branch 'branch-add-gradle-support'

yespiggy committed:
Add Some JUnit Tests

yespiggy committed:
Merge JUnit branch to master

yespiggy committed:
Update build.gradle to package the app as an executable JAR file

yespiggy committed:
Add a way to find tasks by searching for a keyword

yespiggy committed:
merge Level-9 Find into master branch

yespiggy committed:
Improve coding standards

yespiggy committed:
Add JavaDoc comments except for getters and setters

yespiggy committed:
Fix the merge conflict after merging branch-A-JavaDoc

yespiggy committed:
Merge branch-A-CodingStandard to master

yespiggy committed:
Fix bugs occured after updaing getSize() in Task class

yespiggy committed:
Level 10. Add a GUI to the chatbot

yespiggy committed:
Merge chatbot GUI to master branch

JukeBoxK commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+public class Deadline extends Task {
+
+    /** Due dtae of deadline */
+    protected String by;
```
Should `by` be a more specific name like `byDate`? Perhaps you could think of a more specific name.

JukeBoxK commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,189 @@
+     * @throws FileNotFoundException For cases when file is not found.
+     */
+    public ArrayList<Task> loadFile() throws FileNotFoundException {
+        ArrayList<Task> ls = new ArrayList<>();
```
Could `ls` be longer? The name might be too short for others to understand at a quick glance.

JukeBoxK commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,189 @@
+     */
+    public ArrayList<Task> loadFile() throws FileNotFoundException {
+        ArrayList<Task> ls = new ArrayList<>();
+        ArrayList<String> existing = new ArrayList<>();
```
Should `existing` have more details? I believe this is for existing list, maybe `existingLists` might be better.

JukeBoxK commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+public class TaskList {
+
+    /** An arraylist of task for TaskList */
+    private ArrayList<Task> ls;
```
Same question here. Could `ls` be longer?

JukeBoxK commented on src/main/resources/images/DaDog.jpeg:
I like this image.

JukeBoxK commented on src/main/resources/images/DaMorningDog.jpeg:
This is a nice image.

JukeBoxK reviewed (commented):
Great job on ensuring that all method names are verbs. There is a small problem with some variable name not being descriptive enough, but other than that, the code is great in terms of Naming Standard.

menatahari commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Deadline extends Task {
+
+    /** Due dtae of deadline */
```
There's a minor typo in the typing of date.

menatahari commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,33 @@
+
+public class Event extends Task{
+    /** Start time of event */
+    protected String from;
+    /** End time of event */
+    protected String to;
```
Perhaps you can use a more intuitive noun as the variable name, like start or startTime and end or endTime?

menatahari commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,189 @@
+            String tk = existing.get(i);
+            String[] arr = tk.split("<>", 4);
+
+            switch (arr[0]) {
+                case "T":
+                    try {
+                        if (Integer.valueOf(arr[1]) == 0) {
+                            //The task is not finished
+                            Task t = new ToDo(arr[2]);
+                            ls.add(t);
+                        } else if (Integer.valueOf(arr[1]) == 1) {
+                            //Task Finished
+                            Task t = new ToDo(arr[2]);
+                            t.mark();
+                            ls.add(t);
+                        } else {
+                            Ui.botUnknownFormat(i);
+                        }
+                    } catch (Exception e) {
+                        Ui.botUnknownFormat(i);
+                    }
+                    break;
```
According to the coding standard, case clauses in switch statements should not be indented. Perhaps you can configure your IDE to follow this style?

menatahari commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,57 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public boolean getStatus() {
+        return isDone;
+    }
```
Perhaps you can change the method name to sound more boolean?

menatahari commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,157 @@
+    /**
+     * Inform users that the description entered is invalid.
+     */
+    public String invalidDescription() {
+        return "Oops! Sorry, I don't know what that means. Description is empty";
+    }
```
As this is a method, perhaps you can change the name to a verb? I also noticed the same issue in a few other methods.

menatahari commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+     * @param taskNumtoRemove The task number of the to-be-deleted task.
+     */
+    public void remove(int taskNumtoRemove) {
+        getLs().
+                remove(taskNumtoRemove);
```
The statement doesn't look too long. Could this be written in one line?
If you want to keep it as two lines, perhaps you can break the line before the dot separator symbol.

menatahari reviewed (commented):
Looks good overall! Just some minor things to double-check and fix.

yespiggy committed:
Long method in Duke.java

The getResponse method in Duke.java is too long and nested.

Extracting some parts out will improve code readability.

Some parts of code in getResponse is abstracted out into
either Parser or Ui.

yespiggy committed:
Update name of application to LilyBot

The original name for this app Duke is not changed.

To make the names in the app more consistent, all related name
is updated to LilyBot.

For most of the methods or classes, the names are updated by
refactoring, which effectively update all related fields.

yespiggy committed:
Add assertions into LilyBot

For some commands entered by the user, the getResponse
method assumes that the taksNumber entered is a positive
integer.

This is an assumption in the method.

Assertions are used here to check whether this assumption
is true here.

yespiggy committed:
Improve code quality in Storage class

Storage class and LilyBot class may violate expecting
code quality.

Some code quality violations include not following the
SLAP principle, using long nested methods, and unused code
chuncks.

Improve code quality can make the code more easily understandable.

yespiggy committed:
Merge pull request #2 from yespiggy/branch-A-CodeQuality

Improve code quality in Storage class

yespiggy committed:
Merge branch 'master' into branch-A-Assertions
Sync branch-A-Assertions with master branch

yespiggy committed:
Merge pull request #1 from yespiggy/branch-A-Assertions

Add assertions into LilyBot

yespiggy committed:
Correct error in event storage & Use streams

yespiggy committed:
Add extension of undo

yespiggy committed:
merge extension branch to master

yespiggy committed:
Update header comment and code format

yespiggy committed:
Add title bar for LilyBot

yespiggy committed:
Update user guide and add product screenshot

yespiggy committed:
Update README.md

yespiggy committed:
Update GUI: exit when say bye

yespiggy committed:
Update README.md

yespiggy committed:
Update build.gradle

yespiggy committed:
Update jar file download link in README

yespiggy committed:
More oop and fix greeting message

yespiggy committed:
Merge branch of more oop

yespiggy committed:
Update junit tests and command parser

yespiggy committed:
fix: add checkstyle and fix it

yespiggy committed:
fix: add checkstyle and fix it

yespiggy committed:
Merge updated master branch into branch-more-oop

yespiggy committed:
fix methods that are too long

yespiggy committed:
fix error in Storage

yespiggy committed:
fix test for event

yespiggy committed:
fix nested and long method in undo command

yespiggy committed:
More OOP: make commands more logical

Change the constructor for command to include only Ui and TaskList.

Command constructor with these two objects are more reasonable. Also,
two tests are added for EventTest for handling exceptions.

yespiggy committed:
fix checkstyle and some tests

yespiggy committed:
Merge branch-more-oop into master
